### PR TITLE
[TreeView] Remove `instance.getTreeItemIdAttribute`

### PR DESCRIPTION
**Pull request by flaviendelangle**
Extracted from #14210

This is typically the type of instance methods that require us to re-render every item since it's used in the render of the item.
We need to make sure the id attribute of every item is up to date at all time, but with this PR we only have to listen to `treeId` in order to update the id attribute of the item.

### CODE REVIEW

File: packages/x-tree-view/src/internals/corePlugins/useTreeViewId/useTreeViewId.types.ts [hunk 1]
```diff
@@ -23,8 +10,19 @@ export interface UseTreeViewIdParameters {
 
 export type UseTreeViewIdDefaultizedParameters = UseTreeViewIdParameters;
 
+export interface UseTreeViewIdState {
+  id: {
```
Comment (noraleonte) [inline packages/x-tree-view/src/internals/corePlugins/useTreeViewId/useTreeViewId.types.ts:14]: Quick question: any reason why we are nesting `treeId`? Is it to be able to identify the state related to a certain plugin more easily? Should we do the same for other plugin states?

**Review by noraleonte — APPROVED** [review]
Great work! 🚀

File: packages/x-tree-view/src/internals/corePlugins/useTreeViewId/useTreeViewId.types.ts (hunk 1 above)
Comment (noraleonte) [reply packages/x-tree-view/src/internals/corePlugins/useTreeViewId/useTreeViewId.types.ts:14]: `UseTreeViewFocusState` and `UseTreeViewLabelState` do not follow this convention from what I see

Comment (flaviendelangle) [reply packages/x-tree-view/src/internals/corePlugins/useTreeViewId/useTreeViewId.types.ts:14]: I think I fixed those in #14210 :angel: 
But that's something I should definitely extract

**Pull request merged**